mh/pat_item.py

```diff
@@ -693,61 +693,62 @@ class CircleInfo(PatData):
 
     @staticmethod
     def pack_from(circle, circle_index):
-        circle_info = CircleInfo()
-        circle_info.index = Long(circle_index)
+        with circle.lock():
```

Comment:
Can't it be locked where it's going to be packed rather than in the `pack_from` function?

Reply:
All situations where the `pack_from` method is called will require a lock over the whole thing, so wouldn't it be simpler to just have the `pack_from` methods themselves contain the lock? That way we don't need to worry about doing the locking every time we call these methods.

Reply:
I agree with @kbarkevich, maybe making the locking more explicit? Another method called: `lock_and_pack_from`

Reply:
The current approach should be safe as the lock is reentrant. I'll merge it as is for the time being.